earth_enterprise/src/fusion/asset-tests/include/mem_usage/access_private_variables.hpp

```diff
@@ -0,0 +1,77 @@
+#ifndef _MEM_USAGE_ACCESS_PRIVATE_VARIABLES_HPP
+#define _MEM_USAGE_ACCESS_PRIVATE_VARIABLES_HPP
+
+
+namespace mem_usage {
+
+namespace access_private_variables {
+
+// From <https://gist.github.com/dabrahams/1528856>:
+
+// This is a rewrite and analysis of the technique in this article:
+// http://bloglitb.blogspot.com/2010/07/access-to-private-members-thats-easy.html
+
+// ------- Framework -------
+// The little library required to work this magic
+
+// Generate a static data member of type Tag::type in which to store
+// the address of a private member.  It is crucial that Tag does not
+// depend on the /value/ of the the stored address in any way so that
+// we can access it from ordinary code without directly touching
+// private data.
+template <class Tag>
+struct stowed
+{
+     static typename Tag::type value;
+}; 
+template <class Tag> 
+typename Tag::type stowed<Tag>::value;
+
+// Generate a static data member whose constructor initializes
+// stowed<Tag>::value.  This type will only be named in an explicit
+// instantiation, where it is legal to pass the address of a private
+// member.
+template <class Tag, typename Tag::type x>
+struct stow_private
+{
+     stow_private() { stowed<Tag>::value = x; }
+     static stow_private instance;
+};
+template <class Tag, typename Tag::type x> 
+stow_private<Tag,x> stow_private<Tag,x>::instance;
+
```

Comment:
minor, nomenclature: perhaps use "store" instead of "stow"

Reply:
This is another author's code.  Maybe he prefers British, or Australian, or other English.  I'm not sure it's worth changing.